examples/rust/components/blobby/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+            }
+        }
+    }
+}
+
+// HACK(thomastaylor312): We are returning the full object in memory because there isn't really a
+// way to glue in the streams to each other right now. This should get better in wasi 0.2.2 with the
+// stream forward function
+fn get_object(container_name: &String, object_name: &String) -> Result<Vec<u8>> {
+    // Check that the object exists first. If it doesn't return the proper http response
+    let container =
+        blobstore::get_container(container_name).map_err(Error::from_blobstore_error)?;
+    if !container
+        .has_object(object_name)
+        .map_err(Error::from_blobstore_error)?
+    {
+        return Err(Error::not_found());
+    }
+
+    let metadata = container
+        .object_info(object_name)
+        .map_err(Error::from_blobstore_error)?;
+    let incoming = container
+        .get_data(object_name, 0, metadata.size)
+        .map_err(Error::from_blobstore_error)?;
+    let body = wasi::blobstore::types::IncomingValue::incoming_value_consume_sync(incoming)
+        .map_err(Error::from_blobstore_error)?;
+
+    log(Level::Info, "get_object", "successfully got object stream");
+    Ok(body)
+}
+
+fn delete_object(container_name: &String, object_name: &String) -> Result<StatusCode> {
+    let container =
+        blobstore::get_container(container_name).map_err(Error::from_blobstore_error)?;
+
+    container
+        .delete_object(object_name)
+        .map(|_| StatusCode::OK)
+        .map_err(Error::from_blobstore_error)
+}
+
+// HACK(thomastaylor312): We are passing the full object in memory because there isn't really a way
+// to glue in the streams to each other right now. This should get better in wasi 0.2.2 with the
+// stream forward function
+fn put_object(container_name: &String, object_name: &String, data: Vec<u8>) -> Result<StatusCode> {
+    let container =
+        blobstore::get_container(container_name).map_err(Error::from_blobstore_error)?;
+    let result_value = wasi::blobstore::types::OutgoingValue::new_outgoing_value();
+
+    let mut body = result_value
+        .outgoing_value_write_body()
+        .expect("failed to get outgoing value output stream");
+
+    let mut out = OutputStreamWriter::from(&mut body);
+
+    if let Err(e) = out.write_all(&data) {
+        log(
+            Level::Error,
+            "put_object",
+            &format!("Failed to write data to blobstore: {}", e),
+        );
+        return Err(Error {
+            status_code: StatusCode::INTERNAL_SERVER_ERROR,
+            message: format!("Failed to write data to blobstore: {}", e),
+        });
+    }
+
+    if let Err(e) = out.flush() {
+        log(
+            Level::Error,
+            "put_object",
+            &format!("Failed to flush data to blobstore: {}", e),
+        );
+        return Err(Error {
+            status_code: StatusCode::INTERNAL_SERVER_ERROR,
+            message: format!("Failed to flush data to blobstore: {}", e),
+        });
+    }
```

Comment:
why not use `blocking-splice` instead? I don't think `forward` will be added to streams, since `splice` already exists

Reply:
Blocking splice wasn't working for me and it required me to write a bunch of looping stuff and know the content length to read. This was simple enough to get started and then we can go back and optimized

Reply:
Oh and I did check with @ricochet and she says she's pretty sure forward is coming in. We can circle back on this and improve in a follow up